### PR TITLE
Expose further configuration options

### PR DESCRIPTION
**Pull request by rovo89**
See https://github.com/sahlberg/libnfs/pull/398.

### CODE REVIEW

**Comment by rovo89** [comment]
https://github.com/sahlberg/libnfs/pull/398 has already been merged. Not sure how to retrigger the checks, I thought they wouldn't run if I create the PR as a draft...

**Comment by FGasper** [comment]
@rovo89: Generally to rerun checks I just recreate the top commit and force-push. ("Brute force", heh.)

**Comment by rovo89** [comment]
Realized that the submodule had to be updated and that there's no release including the new commit yet... Seems like the Mac and BSD tests failed due to the latter. Also not sure if you want to reference an unreleased commit, so probably it's better to just let this PR sit here until a new upstream version has been released.

**Comment by FGasper** [comment]
@rovo89 Can you please add some defines to avoid breaking compat with existing dynamic libnfs installs?

Since (AFAICT) libnfs doesn’t expose a means to discover various features directly, it might involve ExtUtils::CBuilder or some such in Makefile.PL.

**Comment by rovo89** [comment]
Phew... you're right of course, however I never did such things before. Will have to see how that stuff works. This would be a compile-time check, right? So if people compile the module on a host with the latest commits and then take the lib to a server with libnfs 5.0.2, it'll not work/crash?

**Comment by FGasper** [comment]
Sort of … the idea is that if the libnfs we’re using—whether static or dynamic—exposes your new functions, then we can use them; otherwise we can’t.

Right now, this PR breaks compatibility with existing libnfs releases. I considered actually building more feature detection in so that Net::LibNFS could be used against pre-v5 libnfs releases, but I wasn’t sure how useful that would be.

**Comment by FGasper** [comment]
https://github.com/FGasper/p5-Net-LibNFS/blob/aux_gids_upstream/Makefile.PL
^^ Example of the feature-checking.

**Comment by FGasper** [comment]
@rovo89 I ported my feature-detection logic into a new branch based off yours: https://github.com/FGasper/p5-Net-LibNFS/tree/rovo89_more_opts

If it smokes and looks good to you, I’ll merge.

**Comment by rovo89** [comment]
Looks good, thanks! I won't be able to actually test it before Friday though, sorry.

File: LibNFS.xs [hunk 1]
```diff
@@ -1178,6 +1205,14 @@ set (SV* self_sv, ...)
 
                 continue;
             }
+#ifdef NLNFS_NFS_SET_READDIR_MAX_BUFFER_SIZE
```
Comment (rovo89) [inline LibNFS.xs:1208]: Should be one line below, but then `xsubpp` complains that "Code is not inside a function".

**Comment by rovo89** [comment]
> Looks good, thanks! I won't be able to actually test it before Friday though, sorry.

It's Friday today :) Finally got around to test it. I didn't have a recent shared library installed, so I had to copy your code to the static build branch for my tests. It works as expected, i.e. if one feature is not available compilation still works and runtime croaks with "Unknown setting".

I also added support for the new `readdir-buffer` option that was merged yesterday. It's can drastically improve performance, see https://github.com/sahlberg/libnfs/pull/396 and https://github.com/sahlberg/libnfs/pull/404.

Again some tests are failing, unrelated to my changes though as far as I can see...
- FreeBSD fails in `ioasync_badmount.t` with "nfs_mount_async failed!", not sure what that is.
- Cygwin fails while trying to install some packages.
- Windows fails because it doesn't know iovectors which were recently introduced upstream. Not sure if this is a configuration issue or an upstream issue, could you have a look please?

File: LibNFS.xs (hunk 1 above)
Comment (FGasper) [reply LibNFS.xs:1208]: Yeah I think inside an XSUB I’ve done plain C `if (NLNFS_NFS_SET_READDIR_MAX_BUFFER_SIZE) { … }`, relying on the compiler to optimize away `if (1)` or `if (0)`.

**Comment by FGasper** [comment]
I suggest `readdir_buffer` be a separate PR.

Cygwin’s GitHub Action is broken currently; I don’t know when it’ll be back.
Windows … that may be an upstream issue?

**Comment by rovo89** [comment]
Ah, you have already merged the changes.. closing this one then.

**Pull request closed without merging**